### PR TITLE
buildContextExtentionWithUsageInHomeScreenCreated

### PR DESCRIPTION
**Pull request by Jorjo77**
I'm not sure if it is the right usage.

### CODE REVIEW

File: lib/extensions/build_context_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+
+extension BuildContextExtensions on BuildContext {
+   // Convert width percentage to pixel value
+  double widthPercentage(double percentage) {
+    assert(percentage >= 0 && percentage <= 100);
+    return MediaQuery.of(this).size.width * (percentage / 100);
```
Comment (HristoPetkov94) [inline lib/extensions/build_context_extensions.dart:7]: MediaQuery.of(this).size.width to be extracted into private class variable

File: lib/extensions/build_context_extensions.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  }
+  // Convert height percentage to pixel value
+  double heightPercentage(double percentage) {
+    assert(percentage >= 0 && percentage <= 100);
+    return MediaQuery.of(this).size.height * (percentage / 100);
```
Comment (HristoPetkov94) [inline lib/extensions/build_context_extensions.dart:12]: (percentage / 100) this to be separate method  example _getDecimalPercentage(percent)

File: lib/extensions/build_context_extensions.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+
+extension BuildContextExtensions on BuildContext {
+   // Convert width percentage to pixel value
+  double widthPercentage(double percentage) {
+    assert(percentage >= 0 && percentage <= 100);
+    return MediaQuery.of(this).size.width * (percentage / 100);
+  }
+  // Convert height percentage to pixel value
+  double heightPercentage(double percentage) {
```
Comment (HristoPetkov94) [inline lib/extensions/build_context_extensions.dart:10]: method name to be setHeight

File: lib/extensions/build_context_extensions.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+
+extension BuildContextExtensions on BuildContext {
+   // Convert width percentage to pixel value
+  double widthPercentage(double percentage) {
+    assert(percentage >= 0 && percentage <= 100);
```
Comment (HristoPetkov94) [inline lib/extensions/build_context_extensions.dart:6]: test what happens if you add number out of the scope 0 and 100

File: lib/extensions/build_context_extensions.dart [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+
+extension BuildContextExtensions on BuildContext {
+   // Convert width percentage to pixel value
```
Comment (HristoPetkov94) [inline lib/extensions/build_context_extensions.dart:4]: redundant comment

**Review by HristoPetkov94 — CHANGES_REQUESTED** [review]

**Comment by Jorjo77** [comment]
I'm not sure, but maby waiting for approvement...

**Review by HristoPetkov94 — APPROVED** [review]

**Pull request merged**